packages/phoenix-client/src/phoenix/client/utils/template_formatters.py

```diff
@@ -121,11 +121,9 @@ def _format(
         variables: Mapping[str, str] = MappingProxyType({}),
     ) -> str:
         for variable_name in variable_names:
-            template = re.sub(
-                pattern=rf"(?<!\\){{{{\s*{variable_name}\s*}}}}",
-                repl=variables[variable_name],
-                string=template,
-            )
+            pattern = rf"(?<!\\){{{{\s*{re.escape(variable_name)}\s*}}}}"
+            replacement = variables[variable_name]
+            template = re.sub(pattern, lambda _: replacement, template)
```

Comment:
how should this be changing behavior? can you add a test to demonstrate the change you expect (and show us that it failed prior to the change)? To my eyes this looks like it should generally behave identically to the old code.

2nd, please invoke `re.sub` with named kwargs, e.g. `pattern=, repl= and string=` for clarity

Reply:
@anticorrelator Thanks for the review! I've added unit tests for the formatters, including a specific test case `mustache-escaped-sequence` to cover the escape sequence mentioned in the issue. This case would fail with the previous implementation.

A similar pattern is used in [this file](https://github.com/Arize-ai/phoenix/blob/main/src/phoenix/utilities/template_formatters.py#L86) as well. If the current changes look good to you, I can update that instance too and add corresponding unit tests.

LMK, if any further changes are required